src/components/pages/ShopDetail.tsx

```diff
@@ -170,11 +206,29 @@ function ShopDetail() {
                 flexDirection: "column"
               }}
             >
-              <CardHeader
-                className="card"
-                title={shop?.name}
-                // subheader={shop.address}
-              />
+              {loggedInStatus === "未ログイン" ?
+                <CardHeader
+                  className="card"
+                  title={shop?.name}
+                  // subheader={shop.address}
+                />
+              :
+                <CardHeader
+                  className="card"
+                  title={shop?.name}
+                  action={
+                    <>
+                      <IconButton 
+                        aria-label="settings"
+                        onClick={favoriteData === defaultFavoriteData ? handleFavoriteClick : handleDeleteFavoriteClick}
```

Comment:
favoriteData と defaultFavoriteData は異なるインスタンスなので常に false になってしまいそうですね！
```suggestion
                        onClick={favoriteData.id === defaultFavoriteData.id ? handleFavoriteClick : handleDeleteFavoriteClick}
```